### DISCUSSION TITLE
Questions about Offscreen.

### DISCUSSION THREAD

**Discussion by Lydanne** [fltk-rs/fltk-rs#1232] [Q&A] [answered]
hello
When I was researching this [demo](https://github.com/fltk-rs/fltk-rs/blob/master/fltk/examples/paint.rs), I came across the Offscreen widget, but I couldn't find more detailed information about it. I wanted to know what this widget does.

**Comment by MoAlyousef** [comment, marked as answer]
Hello
Offscreen is like an invisible canvas. It allows you to draw on it and when your image/drawing is ready, you can render it on top of a widget. Another advantage is that sometimes you need to draw something when handling an event, and since you can’t set the Widget’s draw method during event handling, you can instead draw to an offscreen, then just ask the widget to copy whatever is on the offscreen when the widget is redrawn (like in the paint example).

**Reply by Lydanne** [reply]
I understand, that is, Offscreen can draw anywhere, and then paste the drawing on the corresponding widget when widget.draw, right?

**Reply by MoAlyousef** [reply]
Yes that’s right.
